### ISSUE TITLE
auro-input: v1.3 increased bundle size 5x

### ISSUE THREAD

**Issue by geoffrich** [AlaskaAirlines/auro-input#36]
## Describe the bug
Due to including credit card icons, the bundled size of auro-input is now [156.4 KB minified](https://bundlephobia.com/result?p=@alaskaairux/auro-input@1.3.3). v.1.2 (before the icons) is only [38.3 KB minified](https://bundlephobia.com/result?p=@alaskaairux/auro-input@1.2.0), and almost all of that is LitElement/HTML (which would be shared among other Auro elements used on the page). 

This is almost a 5x increase in bundle size. 90kb of the increase is the icons.

I don't know if this should be considered a bug or not, but "feature" didn't seem to fit.

## To Reproduce
View bundle size analysis at bundlephobia (links above)

## Expected behavior
We should figure out a way to decrease the bundle size. Some increase is inevitable, but this much increase will come as a surprise to those who use the component, especially if they don't need the new features.

Some options:
- optimize the SVGs. The icons are small and don't need the level of precision the current SVGs include.
- don't load the SVGs/Cleave unless the current props require it
- create a separate component, e.g. auro-cc-input, so that only consumers who need the credit card functionality import it (breaking change)

I'm open to other ideas, but 156KB seems waaaaay too big for a text input.

**Comment by geoffrich** [comment]
There also might be an opportunity to build tooling to alert us to these bundle increases so we know how a given PR affects bundle size.

**Comment by blackfalcon** [comment]
Yeah, I was afraid of this. The SVGs for the CC are pretty complex and we embedded them for speed. The only way to bring down this bundle size is to off-load the loading the SVGs to the auro-icon, but there is a perceived performance issue there with getting the SVGs from a CDN. 

I updated the label to be `question` versus bug as the update worked as intended. An update would be a feature update.

**Comment by geoffrich** [comment]
I agree that using auro-icon for the SVGs is not a good idea.

What are your thoughts on making credit card input a separate component? This would definitely be a breaking change, but it would make it simple to only import the credit card icons when the user requires a credit card input. We can share validation behavior with the base input class.

Pros:
- Regular text input use cases (probably the majority of auro-input uses) aren't penalized with 118KB (icons + cleave) of JS that isn't being used
- We maintain 1-1 mapping between `type` and actual HTML input `type`s

Cons:
- Breaking change
- Could be confusing why some inputs are different `type`s and credit card is a separate component

There might also be an opportunity to use dynamic import to only load the icons on demand, though idk how that will play with people's bundling setups. As I mentioned above, can't we also optimize the SVGs somehow using a tool [like this](https://jakearchibald.github.io/svgomg/)? I can't imagine we need the level of detail in the current SVGs when they're only being displayed at 24x24.

Either way, I think we need to do something to mitigate the performance impact. Only speaking for my team, 118KB is a *significant* increase when our entire app payload is 128KB (uncompressed). We still need to update to auro-input, but the performance impact would cause us to delay it or lock to v.1.2 until we figure out a way around this.

**Comment by blackfalcon** [comment]
There is a series of work to be done with auro-input and breaking this out into a sub-component is something that I am in favor of. 

But where I am having a conflict is. If a user needs auro-input both with and without CC support in the same view, if this is NOT a bundle, they will be duplicating code in that experience. 

Whereas if the user does NOT want CC support, having the extra weight is not wanted. I am curious as to your thoughts about having an element that does and does not support CC as a type?

**Comment by geoffrich** [comment]
I think if most of the common code can be moved to the base input, duplicated code should be minimal. If they import `auro-input` and (say) `auro-cc-input`, the shared base-input code should only appear in the final bundle once. This is how the drawer and dialog work.

To verify this, I tried bundling the dialog and drawer in a Svelte app with Rollup. When I only imported the dialog, the bundle size was 54KB. When I imported both the dialog and the drawer, the bundle size was 55 KB. If we use a similar pattern with the input, we should be able to support using both the text and credit card input with a minimal performance impact (and a significant performance gain for users only importing the text input).

> I am curious as to your thoughts about having an element that does and does not support CC as a type?

I don't understand the question here. Do my comments about duplicated code answer this question or is this about something else?
